### DISCUSSION TITLE
Pallene + LuaJIT compatibility

### DISCUSSION THREAD

**Discussion by The-Lord-of-Owls** [pallene-lang/pallene#635] [Q&A]
Is it currently possible to use this along side LuaJIT instead of Lua 5.1 - 5.4?

If not have there been plants to do this at all? I see no real reason to not use the best possible lua implementation along side it

**Comment by hugomg** [comment]
Not possible yet. Pallene's compiler generates code that's compatible with a very particular Lua version. For example, you can't use it with any 5.4, it would have to be 5.4.7

In theory, someone could teach the Pallene backend how to target more than one Lua version, but it would be a non-trivial project.

**Reply by srijan-paul** [reply]
So any plans of supporting a 5.4.7 that doesn't use the patched Lua version?

**Reply by hugomg** [reply]
Ah yes, that's another issue I forgot to mention. Pallene reaches into the guts of the Lua interpreter, including parts that are not in the public Lua API. Currently this requires a patched version of the Lua interpreter that makes these API public. I'm currently torn on what would be the best way to solve this. 

* New installer that can easily install & update the patched Lua?
* Bundle a copy of all the Lua interpreter inside the shared libraries Pallene builds?
* Stop using shared libraries as the default compilation target, and instead produce executables that bundle their copy of the Lua interpreter?
